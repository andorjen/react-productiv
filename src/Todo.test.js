import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", function () {
    // this is a low-value test, but better than nothing
    render(<Todo 
      id={1}
      title="Test title."
      description="This is a test description."
      priority={1} />);
  });
  // end

  it("has the correct title, description and priority", function () {
    const { container, debug } = render(<Todo 
      id={1}
      title="Test title."
      description="This is a test description."
      priority={1} />);
    const todoHtml = container.querySelector(".Todo");
    debug(todoHtml);
  
    expect(todoHtml).toContainHTML("Test title.");
    expect(todoHtml).toContainHTML("(priority: 1)");
    expect(todoHtml).toContainHTML("This is a test description.")
  });
  // end

  it("matches snapshot", function () {
    const { container } = render(<Todo 
      id={1}
      title="Test title."
      description="This is a test description."
      priority={1} />);
    expect(container).toMatchSnapshot();
  });
  // end