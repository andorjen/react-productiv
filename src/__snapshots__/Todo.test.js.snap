// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        Test title.
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        This is a test description.
      </small>
    </div>
  </div>
</div>
`;
